_drafts/installing-gdb.md

```diff
@@ -0,0 +1,514 @@
+FROM ubuntu:22.04@sha256:bace9fb0d5923a675c894d5c815da75ffe35e24970166a48a4460a48ae6e0d19
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+# install GDB + GCC for ARM
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    gcc-arm-none-eabi \
+    gdb-multiarch
+```
+
+This approach has a few advantages:
+
+- anyone using the same Docker image should be using _exactly_ the same version
+  of the tools
+- the same Docker image can be used in your test / CI infrastructure, which
+  simplifies things greatly
+- Docker images can be used on Linux/Mac/Windows without too much host setup
+
+And a few downsides:
+
+- running Docker images on Mac or Windows is likely going to be lower
+  performance than on Linux, especially for file IO, since on those platforms
+  Docker containers are running in a virtualization environment
+- accessing host data (files) from within the Docker container can be tricky
+- accessing host hardware (USB or Serial ports) can be easy or impossible,
+  depending on host platform
+- maintaining the Docker image itself requires some knowledge of Docker (though
+  there are a lot of good tutorials online)
+
+Plug- Memfault also provides a Docker image for running gdb-multiarch:
+
+- <https://docs.memfault.com/docs/mcu/coredump-elf-with-gdb/#multi-architecture-gdb-using-docker>
+
+### SDK-specific tools
+
+SDK Vendors (and IDE vendors too!) will often package copies of GDB, along with
+other toolchain components, to make it easy to get started with their platform.
+
+For example, Zephyr RTOS provides pre-built toolchains, that include the
+appropriate GDB versions:
+
+- <https://github.com/zephyrproject-rtos/sdk-ng/releases/tag/v0.14.2>
+
+ESP-IDF from Espressif provides similar tool packages:
+
+- <https://github.com/espressif/binutils-gdb/releases/esp-gdb-v11.2_20220715>
+
+Advantages:
+
+- these packages are often well-tested by the vendors, and you may even be able
+  to get support if you encounter issues!
+- generally support all Linux/Mac/Windows these days
+
+Disadvantages:
+
+- often these packages are very tailored to the particular environment (which
+  usually is just fine), including custom patches for example
+- python support can be pretty variable- some packages won't have python support
+  at all, or it will only be for a specific version.
+- external dependencies may not be provided (eg libpython), and you'll have to
+  bring your own
+
+### xPack GNU Arm Embedded GCC
+
+The [xPack Project](https://xpack.github.io/) provides a prebuilt gcc-arm
+toolchain:
+
+- <https://github.com/xpack-dev-tools/arm-none-eabi-gcc-xpack/releases/>
+
+They tend to be pretty up-to-date, and more reliably include python support.
+
+There's also build documentation, if you'd like to tweak the package:
+
+- <https://github.com/xpack-dev-tools/arm-none-eabi-gcc-xpack/blob/xpack/README-BUILD.md>
+
+Overall pretty simple and easy to use, and could be a good alternative to the
```

Comment:
This looks like the best option? May be worth highlighting stronger pros/cons. Currently it reads as "this is another option we haven't dug into much".

Reply:
Yeah, good call! I'll move it under the arm one as a very viable alternative.